dbms/src/Storages/Transaction/SSTReader.h

```diff
@@ -14,28 +14,110 @@
 
 #pragma once
 
+#include <Common/Logger.h>
 #include <Common/nocopyable.h>
 #include <Storages/Transaction/ProxyFFI.h>
 
 namespace DB
 {
-
 struct SSTReader
 {
-    bool remained() const;
-    BaseBuffView key() const;
-    BaseBuffView value() const;
-    void next();
+    virtual bool remained() const;
+    virtual BaseBuffView key() const;
+    virtual BaseBuffView value() const;
+    virtual void next();
 
     DISALLOW_COPY_AND_MOVE(SSTReader);
     SSTReader(const TiFlashRaftProxyHelper * proxy_helper_, SSTView view);
-    ~SSTReader();
+    // Should only be used when SSTReader is a base class.
+    SSTReader();
+    virtual ~SSTReader();
 
 private:
     const TiFlashRaftProxyHelper * proxy_helper;
     SSTReaderPtr inner;
     ColumnFamilyType type;
+    bool inited;
 };
 
+/// MultiSSTReader helps when there are multiple sst files in a column family.
+/// It is derived from virtual class SSTReader, so it can be holded in a SSTReaderPtr.
+/// It also maintains instance of `R` which is normaly SSTReader(and MockSSTReader in tests),
+/// to read a single sst file.
+/// An `Initer` function is need to create a instance of `R` from a instance of `E`,
+/// which is usually path of the SST file.
+/// We introduce `R` and `E` to make this class testable.
+template <typename R, typename E>
+struct MultiSSTReader : SSTReader
+{
+    using Initer = std::function<std::unique_ptr<R>(const TiFlashRaftProxyHelper *, E)>;
+
+    DISALLOW_COPY_AND_MOVE(MultiSSTReader);
+
+    bool remained() const override
+    {
+        this->maybe_next_reader();
+        return mono->remained();
+    }
+    BaseBuffView key() const override
+    {
+        return mono->key();
+    }
+    BaseBuffView value() const override
+    {
+        return mono->value();
+    }
```

Comment:
Suggest adding "View" suffix to these two methods as they return "BaseBuffView". The same for class `SSTReader`.
```suggestion
    BaseBuffView keyView() const override
    {
        return mono->key();
    }
    BaseBuffView valueView() const override
    {
        return mono->value();
    }
```